addons/membership/models/account_move.py

```diff
@@ -76,6 +72,21 @@ def write(self, vals):
                 'account_invoice_line': line.id,
             })
         self.env['membership.membership_line'].create(memberships_vals)
+        for membership in existing_memberships:
+            line = membership.account_invoice_line
+            date_from = line.product_id.membership_date_from
+            date_to = line.product_id.membership_date_to
+            if (
+                date_from and date_from < (line.move_id.invoice_date or date.min) < (date_to or date.min)
+            ):
+                date_from = line.move_id.invoice_date
+            membership.write({
+                "membership_id": line.product_id.id,
+                'member_price': line.price_unit,
+                'date': fields.Date.today(),
+                'date_from': date_from,
+                'date_to': date_to,
+            })
```

Comment:
Dates and payload computation is repeat 3 times in this files, I tried to avoid to change in existing code but I'm tempt to factorised common data generation, this will also helps overwrite it in sub modules :thinking:

Reply:
I've refactored a bit hopping Odoo accept the change.... :crossed_fingers: